instrumentation/rxjava-1.0/src/main/java/io/opentelemetry/instrumentation/auto/rxjava/TracedOnSubscribe.java

```diff
@@ -16,56 +16,44 @@
 
 package io.opentelemetry.instrumentation.auto.rxjava;
 
-import static io.opentelemetry.trace.TracingContextUtils.currentContextWith;
-
-import io.opentelemetry.OpenTelemetry;
+import io.grpc.Context;
+import io.opentelemetry.context.ContextUtils;
 import io.opentelemetry.context.Scope;
-import io.opentelemetry.instrumentation.api.decorator.BaseDecorator;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
 import io.opentelemetry.trace.Span;
-import io.opentelemetry.trace.Tracer;
 import rx.Observable;
 import rx.Subscriber;
 import rx.__OpenTelemetryTracingUtil;
 
 public class TracedOnSubscribe<T> implements Observable.OnSubscribe<T> {
-  private static final Tracer TRACER = OpenTelemetry.getTracer("io.opentelemetry.auto.rxjava-1.0");
-
-  private final Observable.OnSubscribe<?> delegate;
-  private final String operationName;
-  private final Span parentSpan;
-  private final BaseDecorator decorator;
-  private final Span.Kind spanKind;
+  protected final Observable.OnSubscribe<?> delegate;
+  protected final String operationName;
+  protected final Context parentContext;
+  protected final BaseTracer tracer;
+  protected final Span.Kind spanKind;
 
   public TracedOnSubscribe(
-      Observable originalObservable,
-      String operationName,
-      BaseDecorator decorator,
-      Span.Kind spanKind) {
+      Observable originalObservable, String operationName, BaseTracer tracer, Span.Kind spanKind) {
     delegate = __OpenTelemetryTracingUtil.extractOnSubscribe(originalObservable);
     this.operationName = operationName;
-    this.decorator = decorator;
+    this.tracer = tracer;
     this.spanKind = spanKind;
 
-    parentSpan = TRACER.getCurrentSpan();
+    parentContext = Context.current();
   }
 
   @Override
   public void call(Subscriber<? super T> subscriber) {
-    // span finished by TracedSubscriber
-    Span.Builder spanBuilder = TRACER.spanBuilder(operationName).setSpanKind(spanKind);
-    if (parentSpan != null) {
-      spanBuilder.setParent(parentSpan);
-    }
-    Span span = spanBuilder.startSpan();
-
-    afterStart(span);
-
-    try (Scope scope = currentContextWith(span)) {
-      delegate.call(new TracedSubscriber(span, subscriber, decorator));
+    try (Scope ignored = ContextUtils.withScopedContext(parentContext)) {
```

Comment:
i think good to prefer not creating unnecessary scopes

though it would require a `BaseTracer.startSpan` overload that takes a parent Context (or Span)

Reply:
Yeah, exactly. I will just a TODO here for now and in the future I will review all `startSpan` methods anyway. They are kinda chaotic right now.